packages/matrix/tests/messages.spec.ts

```diff
@@ -352,39 +354,50 @@ test.describe('Room messages', () => {
         },
       },
       {
+        type: 'function',
         function: {
-          description: `Propose a query to search for a card instance filtered by type. Always prioritise search based upon the card that was last shared.`,
           name: 'searchCard',
+          description:
+            'Propose a query to search for a card instance filtered by type.   If a card was shared with you, always prioritise search based upon the card that was last shared.   If you do not have information on card module and name, do the search using the `_cardType` attribute.',
           parameters: {
+            type: 'object',
             properties: {
               description: {
                 type: 'string',
               },
               filter: {
+                type: 'object',
                 properties: {
                   type: {
+                    type: 'object',
                     properties: {
                       module: {
-                        description: 'the absolute path of the module',
                         type: 'string',
+                        description: 'the absolute path of the module',
                       },
                       name: {
-                        description: 'the name of the module',
                         type: 'string',
+                        description: 'the name of the module',
                       },
                     },
                     required: ['module', 'name'],
+                  },
+                  eq: {
                     type: 'object',
+                    properties: {
+                      _cardType: {
+                        type: 'string',
+                        description: 'name of the card type',
+                      },
+                    },
+                    required: ['_cardType'],
```

Comment:
I think this is ok. You use the prompt to guide the ai to only use eq when not using type. I use the "anyOf" json schema and nest everything into an every: block -- but this is still relying on the ai to get it correct. 

I think im good with this change. And maybe I should just remove the every option here https://github.com/cardstack/boxel/pull/1565/files#diff-81317cc1fb634cb4d63e3ed53a58a3ccf3acf8cd11c041ebb8686291885d12feR429-R443

Reply:
Yea I think the eq and type are redundant. But its ok to extend. 

I think thats a fair point. if you use `anyOf` you need `every` I believe. Lets be conservative and use `oneOf` ? That way we know its a very specific query that the ai is requesting for. But, I wonder if this schema validation is followed by the ai at all (in this respect `every` is better at valid queries). 

If `oneOf` works, we can try that.

Reply:
anyOf I think should be supported but oneOf isn't (though I think with the non strict mode just gets ignored). Is the query valid if both are specified? If so I'd leave it at that, with the non-strict mode the model is able to return any structure it wants. If we have a bunch of issues from it we can also split it into two functions